File: docs/Euler-Particles/color/hsb/hsb_color_generator.js
class HSBColorGenerator {

    randomColor() {
        colorMode(HSB, 360);
        let saturation = int(random(100, 360));
        let brightness = int(random(100, 360));
        let hue = int(random(360));
        let color = this.mapColor(hue, 0, 360, saturation, brightness);
        return color;
    }

    mapColor(value, minValue, maxValue, saturation, brightness) {
        colorMode(HSB, 360);
        return color(0);
    }

}

File: Acceleration/src/color/HSB/HSBColorGenerator.java
package color.HSB;

import processing.core.PApplet;

import color.ColorGenerator;

public abstract class HSBColorGenerator extends ColorGenerator {

    public HSBColorGenerator(PApplet p) {
        super(p);
    }

    public int getRandomColor() {
        p.colorMode(PApplet.HSB, 360);
        int saturation = (int)p.random(100, 360);
        int brightness = (int)p.random(100, 360);
        int hue = (int)p.random(360);
        return mapColor(hue, 0, 360, saturation, brightness);
    }

}
